core/math/vector4.h

```diff
@@ -0,0 +1,463 @@
+
+	_FORCE_INLINE_ Vector4 abs() const;
+	_FORCE_INLINE_ Vector4 floor() const;
+	_FORCE_INLINE_ Vector4 sign() const;
+	_FORCE_INLINE_ Vector4 ceil() const;
+	_FORCE_INLINE_ Vector4 round() const;
+	Vector4 clamp(const Vector4 &p_min, const Vector4 &p_max) const;
+
+	_FORCE_INLINE_ real_t distance_to(const Vector4 &p_to) const;
+	_FORCE_INLINE_ real_t distance_squared_to(const Vector4 &p_to) const;
+
+	_FORCE_INLINE_ Vector4 posmod(const real_t p_mod) const;
+	_FORCE_INLINE_ Vector4 posmodv(const Vector4 &p_modv) const;
+	_FORCE_INLINE_ Vector4 project(const Vector4 &p_to) const;
+
+	_FORCE_INLINE_ Vector4 direction_to(const Vector4 &p_to) const;
+
+	_FORCE_INLINE_ Vector4 bounce(const Vector4 &p_normal) const;
+	_FORCE_INLINE_ Vector4 reflect(const Vector4 &p_normal) const;
+
+	bool is_equal_approx(const Vector4 &p_v) const;
+
+	/* Operators */
+
+	_FORCE_INLINE_ const real_t &operator[](const int p_axis) const;
+	_FORCE_INLINE_ real_t &operator[](const int p_axis);
+
+	_FORCE_INLINE_ Vector4 &operator+=(const Vector4 &p_v);
+	_FORCE_INLINE_ Vector4 operator+(const Vector4 &p_v) const;
+	_FORCE_INLINE_ Vector4 &operator-=(const Vector4 &p_v);
+	_FORCE_INLINE_ Vector4 operator-(const Vector4 &p_v) const;
+	_FORCE_INLINE_ Vector4 &operator*=(const Vector4 &p_v);
+	_FORCE_INLINE_ Vector4 operator*(const Vector4 &p_v) const;
+	_FORCE_INLINE_ Vector4 &operator/=(const Vector4 &p_v);
+	_FORCE_INLINE_ Vector4 operator/(const Vector4 &p_v) const;
+
+	_FORCE_INLINE_ Vector4 &operator*=(const real_t p_scalar);
+	_FORCE_INLINE_ Vector4 operator*(const real_t p_scalar) const;
+	_FORCE_INLINE_ Vector4 &operator/=(const real_t p_scalar);
+	_FORCE_INLINE_ Vector4 operator/(const real_t p_scalar) const;
+
+	_FORCE_INLINE_ Vector4 operator-() const;
+
+	_FORCE_INLINE_ bool operator==(const Vector4 &p_v) const;
+	_FORCE_INLINE_ bool operator!=(const Vector4 &p_v) const;
+	_FORCE_INLINE_ bool operator<(const Vector4 &p_v) const;
+	_FORCE_INLINE_ bool operator<=(const Vector4 &p_v) const;
+	_FORCE_INLINE_ bool operator>(const Vector4 &p_v) const;
+	_FORCE_INLINE_ bool operator>=(const Vector4 &p_v) const;
+
+	operator String() const;
+
+	_FORCE_INLINE_ Vector4() {}
+	_FORCE_INLINE_ Vector4(const real_t p_x, const real_t p_y, const real_t p_z, const real_t p_w) {
+		x = p_x;
+		y = p_y;
+		z = p_z;
+		w = p_w;
+	}
+};
+
+// Multiplication operators required to workaround issues with LLVM using implicit conversion
+// to Vector3i instead for integers where it should not.
+
+_FORCE_INLINE_ Vector4 operator*(const float p_scalar, const Vector4 &p_vec) {
+	return p_vec * p_scalar;
+}
+
+_FORCE_INLINE_ Vector4 operator*(const double p_scalar, const Vector4 &p_vec) {
+	return p_vec * p_scalar;
+}
+
+_FORCE_INLINE_ Vector4 operator*(const int32_t p_scalar, const Vector4 &p_vec) {
+	return p_vec * p_scalar;
+}
+
+_FORCE_INLINE_ Vector4 operator*(const int64_t p_scalar, const Vector4 &p_vec) {
+	return p_vec * p_scalar;
+}
```

Comment:
I tried to remove it, but it wouldn't compile. I have a feeling that this comment might be outdated since it seems like these operators are needed regardless of having an integer variant of Vector4.